pkg/agent/sysadvisor/plugin/qosaware/resource/resource.go

```diff
@@ -62,6 +62,7 @@ type SubResourceAdvisor interface {
 
 type resourceAdvisorWrapper struct {
 	subAdvisorsToRun map[types.QoSResourceName]SubResourceAdvisor
+	metaCache        metacache.MetaCache
```

Comment:
Not recommended to add metaCache here. To align with region info update, it's better to update headroom in xxxResoureAdvisor's GetHeadroom() function.